src/main/java/org/commcare/formplayer/services/NewFormResponseFactory.java

```diff
@@ -68,9 +73,12 @@ public NewFormResponse getResponse(NewSessionRequestBean bean, String postUrl) t
                 bean.getRestoreAs(),
                 bean.getRestoreAsCaseId());
 
+        FormDef formDef = parseFormDef(formXml);
+        String serializedFormDef = FormDefStringSerializer.serialize(formDef);
```

Comment:
This is what I was thinking could be moved into the service layer so we avoid doing the serialisation if we're loading it from the DB

Reply:
See ed0229dc5755f8fb95512c5a60640cafa43ea457